spec/javascripts/controllers/short_form_application_controller_spec.coffee

```diff
@@ -672,3 +677,50 @@ do ->
       it ' calls claimedCustomPreference on ShortFormApplicationService', ->
         scope.claimedCustomPreference()
         expect(fakeShortFormApplicationService.claimedCustomPreference).toHaveBeenCalled()
+
+    describe 'chooseDraft', ->
+      describe 'user chooses recent application and has different account info', ->
+        it 'sends user to choose application details', ->
+          scope.application = {test: 'test'}
+          scope.chosenApplicationToKeep = 'recent'
+          fakeAccountService.loggedInUser = {firstName: 'Test', lastName: 'User'}
+          spyOn(fakeShortFormApplicationService, 'hasDifferentInfo').and.returnValue(true)
+          scope.chooseDraft()
+          expect(state.go).toHaveBeenCalledWith('dahlia.short-form-application.choose-applicant-details')
+
+    describe 'chooseApplicantDetails', ->
+      describe 'user chooses to create account', ->
+        it 'signs out user and sends them to create account page', ->
+          scope.chosenAccountOption = 'createAccount'
+          scope.chooseApplicantDetails()
+          expect(fakeAccountService.signOut).toHaveBeenCalled()
+          expect(state.go).toHaveBeenCalledWith('dahlia.short-form-application.create-account')
+
+      describe 'user chooses to continue as guest', ->
+        it 'signs out user and sends them to the last page of the app', ->
+          scope.application.lastPage = 'name'
+          scope.chosenAccountOption = 'continueAsGuest'
+          scope.chooseApplicantDetails()
+          expect(fakeAccountService.signOut).toHaveBeenCalledWith({preserveAppData: true})
```

Comment:
@kaptron @creativecoder @callaghanc can any of you help me figure out why fakeAccountService isn't stubbing correctly? I am about to rabbit hole on it.

Reply:
@chrisdolendo hm when trying to call spy methods that are meant to resolve + call through it can be tricky, looks like you have it all setup with the `deferred.promise`, etc. However looking at some other examples like account_controller_spec, it uses:

```coffee
        it 'routes user to sign-in', ->
          scope.createAccount()
          scope.$apply()
          expect(state.go).toHaveBeenCalledWith('dahlia.sign-in', {skipConfirm: true, newAccount: true})

```

perhaps `scope.$apply()` is needed to call through the callback?

Reply:
that was the trick! resolved in the continuation of this story for https://www.pivotaltracker.com/story/show/151462100